### ISSUE TITLE
problems configuring NLTK 3.2.2 to use Stanford segmenter

### ISSUE THREAD

**Issue by highpost** [nltk/nltk#1603]
I'm having trouble getting NLTK 3.2.2 configured to use the Stanford NLP segmenter. Most of the problems that I see are CLASSPATH and SLF4J related.

With NLTK, there are three ways to identify where to find the SLF4J and Stanford segmenter jar files:

- CLASSPATH environment variable
- SLF4J and STANFORD_SEGMENTER environment variables
- path_to_slf4j and path_to_jar options with the StanfordSegmenter constructor

I've tried variants of all three approaches (see nltk-ss.py) without success. However, I do have an equivalent shell script that works (see nltk-ss.sh).

BTW, I posted a GitHub issue with the Stanford NLP group and heard back that I should use SLF4J 1.7.12.

[nltk-ss.py.txt](https://github.com/nltk/nltk/files/728960/nltk-ss.py.txt)
[nltk-ss.sh.txt](https://github.com/nltk/nltk/files/728961/nltk-ss.sh.txt)

**Comment by highpost** [comment]
I continued to look at this, and I'm not sure how it's supposed to work. Basicaly, config_java() and java() scaffold a Java command-line and when they do this they construct a -cp argument containing both slf4j-api-1.7.12.jar and stanford-segmenter-3.7.0.jar. But when you include a -cp argument, you override any setting in the CLASSPATH environment variable. See the java(1) man page. So I don't understand how the other JAR files that are necessary for the segmenter get loaded.

I've tested this with a shell script where I create a CLASSPATH environment variable with all the Stanford JAR files and then run a java command line. It works when I leave out the -cp option and it fails when I include the -cp option.
